hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -379,7 +386,8 @@ private HoodieRecord<HoodieMetadataPayload> composeRecord(GenericRecord avroReco
   private Map<Pair<String, FileSlice>, List<String>> getPartitionFileSliceToKeysMapping(final String partitionName, final List<String> keys) {
     // Metadata is in sync till the latest completed instant on the dataset
     List<FileSlice> latestFileSlices =
-        HoodieTableMetadataUtil.getPartitionLatestMergedFileSlices(metadataMetaClient, partitionName);
+        HoodieTableMetadataUtil.getPartitionLatestMergedFileSlices(
```

Comment:
I am thinking we can cache the file slices also similar to how we cache the file readers. I don't see a reason for file slices to change unless there is a change in timeline on which case entire FileSystemView will be refreshed.

Reply:
looks like the FileSystemView (MDFSV) caches the entities and so we are good.